src/Store.cs

```diff
@@ -90,6 +97,10 @@ private static class Native
 
             [DllImport(Engine.LibraryName)]
             public static extern void wasmtime_context_set_epoch_deadline(IntPtr handle, ulong ticksBeyondCurrent);
+            
+            [DllImport(Engine.LibraryName)]
+            [return: MarshalAs(UnmanagedType.IUnknown)]
+            public static extern object wasmtime_context_get_data(IntPtr handle);
```

Comment:
`MarshalAs(UnmanagedType.IUnknown)` is intended for COM interop on Windows, which I don't think is appropriate here. I think you should declare it as returning an `IntPtr`, and then use `GCHandle.FromIntPtr(value).Target!` to get the target object from the `GCHandle`.

See https://github.com/bytecodealliance/wasmtime-dotnet/blob/992efb0baee78bad1b4f9cc0a8a5b363313a63c9/src/Value.cs#L384 for an example.

Reply:
Makes sense, fixed